backend/pkg/api/connect/service/acl/v1/service.go

```diff
@@ -106,6 +110,41 @@ func (s *Service) ListACLs(ctx context.Context, req *connect.Request[v1.ListACLs
 		resources[i] = aclResProto
 	}
 
+	// Filter can be nil if the filter contains no fields that are relevant
+	// for Schema Registry ACLs.
+	srFilter := s.srClientMapper.listACLFilterToDescribeACLSR(req.Msg.Filter)
+	checkSchemaRegistryACLs := srFilter != nil && s.cfg.SchemaRegistry.Enabled
+	if !checkSchemaRegistryACLs {
+		return connect.NewResponse(&v1.ListACLsResponse{Resources: resources}), nil
+	}
+
+	srACLres, err := s.consoleSvc.ListSRACLs(ctx, srFilter)
+	if err != nil {
+		// Handle the case where an older cluster returns "Not found" for Schema Registry ACLs
+		// This means the cluster doesn't support Schema Registry ACLs, so we gracefully continue
+		// without returning an error and just return the Kafka ACLs.
+		if se := (*sr.ResponseError)(nil); errors.As(err, &se) && se.StatusCode == http.StatusNotFound {
+			return connect.NewResponse(&v1.ListACLsResponse{Resources: resources}), nil
+		}
+
+		return nil, apierrors.NewConnectError(
+			connect.CodeInternal,
+			fmt.Errorf("error listing Schema Registry ACLs: %w", err),
+			apierrors.NewErrorInfo(v1.Reason_REASON_REDPANDA_SCHEMA_REGISTRY_ERROR.String()),
+		)
```

Comment:
We should probably create our own helper for constructing rich schema registry errors for ConnectRPC as well. We have one for Kafka API and Redpanda Admin API errors already that convert the returned errors in the best possible ConnectRPC error.

See pkg/api/connect/errors/kafka.go and pkg/api/connect/errors/redpanda.go for example